portalnet/src/types/messages.rs

```diff
@@ -182,7 +182,7 @@ impl Default for PortalnetConfig {
             data_radius: Distance::MAX,
             internal_ip: false,
             no_stun: false,
-            enr_file_location: None,
+            trin_data_dir: PathBuf::default(),
```

Comment:
Well, sorry to be a stickler, but this still feels off. Essentially we're still doing the same thing I was previously against, except this time with `PathBuf::default()` instead of an `Option`

Ultimately, it feels like `PortalnetConfig` is not an appropriate place for this argument, since there is no reasonable default value. 

I think we just remove this from `PortalnetConfig` and pass `node_data_dir` into `Discovery::new()` directly. Thoughts?